### PR TITLE
DSPDC-664 Init codebase

### PR DESCRIPTION
**Pull request by danxmoran**
It's big, but the only file with any differences from clinvar-ingest is the `build.sbt`

### CODE REVIEW

**Review by raaidbroad — APPROVED** [review]

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+val beamVersion = "2.18.0"
```
Comment (danxmoran) [inline build.sbt:1]: Review this file

File: README.md [hunk 2]
```diff
@@ -1,2 +1,39 @@
 # monster-scio-utils
```
Comment (danxmoran) [inline README.md:1]: Review this file

File: utils/msg/src/main/scala/org/broadinstitute/monster/common/msg/JsonParser.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.monster.common.msg
```
Comment (danxmoran) [inline utils/msg/src/main/scala/org/broadinstitute/monster/common/msg/JsonParser.scala:1]: Review this file

File: utils/msg/src/main/scala/org/broadinstitute/monster/common/msg/MsgOperations.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  /**
+    * 'Drill' into a message, following a chain of fields, and return the value
+    * at the end of the chain.
+    *
+    * An exception will be thrown if no value can be extracted. Use `tryRead`
+    * for extracting optional fields.
+    */
+  def read[V: MsgParser](msg: Msg, fieldChain: Seq[String]): V =
```
Comment (danxmoran) [inline utils/msg/src/main/scala/org/broadinstitute/monster/common/msg/MsgOperations.scala:67]: Review from here down (`extract` and `tryExtract` are copied from clinvar)

File: utils/msg/src/test/scala/org/broadinstitute/monster/common/msg/JsonParserSpec.scala [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.monster.common.msg
```
Comment (danxmoran) [inline utils/msg/src/test/scala/org/broadinstitute/monster/common/msg/JsonParserSpec.scala:1]: Review this file

File: utils/msg/src/test/scala/org/broadinstitute/monster/common/msg/MsgOperationsSpec.scala [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    MsgOperations.tryExtract[Long](msg, List("key1", "key2", "key4")) shouldBe None
+  }
+  it should "handle a missing field in the middle of a chain in optional extraction" in {
+    val msg = Obj(
+      Str("key1") -> Obj(Str("key2") -> Obj(Str("key3") -> Arr(Str("foo"), Str("bar"))))
+    )
+    MsgOperations.tryExtract[Long](msg, List("key1", "key4", "key3")) shouldBe None
+  }
+
+  ///// Read tests /////
```
Comment (danxmoran) [inline utils/msg/src/test/scala/org/broadinstitute/monster/common/msg/MsgOperationsSpec.scala:80]: Review from here down (tests above are copied from clinvar)

**Review by snwessel — APPROVED** [review]
Read through everything as best I could and it looks good!

**Pull request merged**